ibis/backends/pyspark/compiler.py

```diff
@@ -12,22 +12,33 @@
 import ibis.common.exceptions as com
 import ibis.expr.datatypes as dtypes
 import ibis.expr.operations as ops
+import ibis.expr.rules as rlz
 import ibis.expr.types as ir
 import ibis.expr.types as types
 from ibis import interval
 from ibis.backends.pandas.execution import execute
-from ibis.backends.spark.datatypes import (
+from ibis.expr.timecontext import adjust_context
+from ibis.util import guid
+
+from .datatypes import (
     ibis_array_dtype_to_spark_dtype,
     ibis_dtype_to_spark_dtype,
     spark_dtype,
 )
-from ibis.expr.timecontext import adjust_context
-from ibis.util import guid
-
 from .timecontext import combine_time_context, filter_by_time_context
 
 
-class PySparkTable(ops.DatabaseTable):
+class SparkUDFNode(ops.ValueOp):
```

Comment:
I am not sure these are needed anymore. Does anything break if we remove this?

Reply:
Seems like those are used in `SparkUDF.__call__`. I don't understand the code of the spark udfs well, but I was wondering if only `SparkPandasUDF` should be left now, and `SparkUDF` was only for the spark backend, and not the pyspark one. Udf tests are failing, so there is something that needs to be done for sure.

If I understand correctly, the next are the public udf functions:
```python
>>> from ibis.spark.udf import udf
>>> @udf.elementwise(input_type=[dt.string], output_type=dt.int64)

>>> from ibis.spark.udf import udf
>>> @udf.elementwise.pandas([dt.string], dt.int64)

>>> from ibis.spark.udf import udf
>>> @udf.reduction(input_type=[dt.string], output_type=dt.int64)
```

Is the first one still supported after removing the spark backend, or can I remove it, and everything that depends on it? Including the classes that you mention.

Reply:
The first one is not supported. (ibis.spark.udf namespace can be totally removed).

The correct namespace for UDF is under `ibis.udf.vectorized`.

Reply:
(Sorry for the late reply, I don't have a very good notification setup..)